report/5.Solution/Scheduling.tex

```diff
@@ -84,14 +84,83 @@ \section{Scheduling}\label{solution:scheduling}
 Keep\_USB\_Alive    & 14                            \\
 Receive\_Data       & 13                            \\
 Move\_Motors        & 28                            \\
-Handle\_Lasers      & --                            \\
-Update\_Display     & 480                           \\
-Stop                & --                            
+Handle\_Lasers      & 30                            \\
+Update\_Display     & 480                           \\                     
 \end{tabular}
 \end{table}\label{table:executionTimes}
 
 Since the execution time of all the tasks totals to a mere 535 microseconds, it was deemed non-essential to include minor cycles, as the tasks all depend on receiving new data from the USB, and this will not occur more than once every millisecond due to hardware limitations.
 Likewise, there is a software limitation when using nxtOSEK, since the interrupt service routine is only run once every millisecond.
 With these considerations, the cyclic execution method with equal minor and major cycles were deemed the best solution, since the implementation of a fixed priority task would add more overhead when changing between tasks and preempting them, rather than just allowing the previous tasks to finish executing.
 
-\todo{Update handle lasers and stop times when they are implemented}
+\subsection{Fixed Priority Scheduling}
+However, as the project progressed, parts of the machine intelligence execution was moved to the NXT for execution.
+This lead to tasks that took longer than 1 millisecond to execute, rendering the use of cyclic executive infeasible, as described in Subsection~\cite{Design:Scheduling}. 
+Due to this, the fixed-priority scheduling model was instead utilized. 
+
+After having added machine learning to the \texttt{Move\_Motors} task the execution times are as follows:
```

Comment:
Føler måske at det her skal rykkes lidt op, men det kan vi lige tage mandag. Tror det giver mere mening at bruge det har som argument for hvorfor vi ikke bruger CE, men hey vi kan sagtens bruge FPS.